src/bank00.asm

```diff
@@ -10045,20 +10044,29 @@ addMoney:
     ld   E, A                                          ;; 00:3d7a $5f
     add  HL, DE                                        ;; 00:3d7b $19
     pop  DE                                            ;; 00:3d7c $d1
-    jr   NC, .set                                      ;; 00:3d7d $30 $03
-    ld   HL, rIE                                       ;; 00:3d7f $21 $ff $ff
-.set:
+    push HL
+    ld   HL, wStatusEffect
+    jr   NC, .set_1
+    set  7, [HL]
+.set_1:
+    bit  7, [HL]
+    pop  HL
+    jr   Z, .set_2
+    ld   A, L
+    sub  A, $9f
+    ld   A, H
+    sbc  A, $86
+    jr   C, .set_2
+    ld   HL, $869f
+.set_2:
     ld   A, H                                          ;; 00:3d82 $7c
     ld   [wMoneyHigh], A                               ;; 00:3d83 $ea $bf $d7
     ld   A, L                                          ;; 00:3d86 $7d
     ld   [wMoneyLow], A                                ;; 00:3d87 $ea $be $d7
     call drawMoneyOnStatusBarTrampoline                ;; 00:3d8a $cd $17 $31
     ret                                                ;; 00:3d8d $c9
-    db   $d5, $fa, $bf, $d7, $57, $fa, $be, $d7        ;; 00:3d8e ????????
-    db   $5f, $7b, $95, $6f, $7a, $9c, $67, $30        ;; 00:3d96 ????????
-    db   $03, $21, $00, $00, $7c, $ea, $bf, $d7        ;; 00:3d9e ????????
-    db   $7d, $ea, $be, $d7, $cd, $17, $31, $d1        ;; 00:3da6 ????????
-    db   $c9                                           ;; 00:3dae ?
```

Comment:
This unused block is a removeMoney routine:
`    push de
    ld a, [wMoneyHigh]
    ld d, a
    ld a, [wMoneyLow]
    ld e, a
    ld a, e
    sub l
    ld l, a
    ld a, d
    sbc h
    ld h, a
    jr nc, jr_000_3da2

    ld hl, $0000

jr_000_3da2:
    ld a, h
    ld [wMoneyHigh], a
    ld a, l
    ld [wMoneyLow], a
    call drawMoneyOnStatusBarTrampoline
    pop de
    ret`

I have been thinking it would be nice to have a `removeMoney` instead of code duplication, but that said:
1. It would be difficult to fit both of these in this one space.
2. It doesn't seem to do what is really needed, specifically, if you don't have enough money it reduces you to zero instead of returning a status.
3. It would probably only be used two places.

Any thoughts on using it or is it more valuable as free space?

Reply:
I don't think it's needed. There are presently two places were money could be removed - scriptOpCodeTakeMoney (used by everything other than shops), and purchaseFromVendor (shops). The way the two functions are structured is different too. The former checks if you have enough money, then takes it away. The latter checks if you have enough money, then does a bunch of other stuff (maybe checking for inventory capacity) before removing the money much later.